packages/near-sdk-js/src/api.ts

```diff
@@ -7,21 +7,21 @@ const EVICTED_REGISTER = U64_MAX - 1n;
 // Interface available in QuickJS
 interface Env {
   // Panic
-  panic_utf8(message: Bytes): never;
+  panic_utf8(message: Uint8Array): never;
 
   // Logging
-  log(message: Bytes): void;
-  log_utf8(message: Bytes): void;
-  log_utf16(message: Bytes): void;
+  log(message: string): void;
+  log_utf8(message: Uint8Array): void;
+  log_utf16(message: Uint8Array): void;
 
   // Read from register
-  read_register(register: Register): string;
+  read_register(register: Register): Uint8Array;
 
   // Storage
-  storage_read(key: Bytes, register: Register): bigint;
-  storage_has_key(key: Bytes): bigint;
-  storage_write(key: Bytes, value: Bytes, register: Register): bigint;
-  storage_remove(key: Bytes, register: Register): bigint;
+  storage_read(key: Uint8Array, register: Register): bigint;
+  storage_has_key(key: Uint8Array): bigint;
+  storage_write(key: Uint8Array, value: Uint8Array, register: Register): bigint;
```

Comment:
If these functions also doing the high level task: decode / encode string into Uint8Array, then user do not have an API to deal with raw bytes. At current form they can compose high level functionality: write string to state by combining storage_write + bytes/encode

> what defines if we have string or Uint8Array here

For apis like `current_account_id()`, we are sure it returned is a string. For apis like `storage_read`, it takes and returns array with any bytes, which is more fit Uint8Array.

> And is it possible to do the conversion in api.ts so users can use strings?

Yes. Discussed further in your next comment

Reply:
Actually user has it, because they can use `env.function()` without API wrapper function. But it's less convinient, I agree.

Maybe we can separate this into 2 issues. Function params and return types. Function params can accept `string | Uint8Array` and return types can be Uint8Array if really necessary. `functionRaw` and 'function' is also an option.

It looks like a purely DevEx problem, let's get input from DevRel team.

Reply:
I feel the `function` and `functionRaw` distinction makes sense. At the same time, haven't seen devs using these functions or asking about them often, so implementing two functions without them being popular feels contradictory. In any case, having a function that accepts a single param type is closer to single resp principle.

Reply:
supporting points:
> ... it's less convinient, I agree.
> functionRaw and 'function' is also an option.
> function and functionRaw distinction makes sense
> having a function that accepts a single param type is closer to single resp principle.

about alternative:
> Actually user has it, because they can use env.function() without API wrapper function

Right now `env` is not exported, and we need more discussion on whether to export it (it doubles host function APIs)

>  Function params can accept string | Uint8Array and return types can be Uint8Array if really necessary

I thinked about it, will still be breaking changes for `function(string | Uint8Array) -> Uint8Array` because return type was string, and if return `string | Uint8Array` based on param type, it's kind of magic.

> At the same time, haven't seen devs using these functions or asking about them often,

Yeah it is not often, I've seen one ask: https://github.com/near/near-sdk-js/pull/195#issuecomment-1293070576

Combining your thoughts, I'll go ahead with ` function` + `functionRaw` (I assume Serhii is okay with both functionRaw and functionBytes and @idea404 prefer functionRaw, and naming of less popular low level API can be changed or alias later)